### DISCUSSION TITLE
Customize/extend directories exposed by repositories service

### DISCUSSION THREAD

**Discussion by NicolasT** [scality/metalk8s#3146] [Ideas]
When using MetalK8s as part of a larger product offering, where next to container images as part of 'solutions' and the package repositories part of MetalK8s also other files should be served through some HTTP service (think e.g., other package repositories), it could be useful to reuse the repositories `nginx` deployed as part of MetalK8s for said purpose.

As such, open for discussion: should it be possible to (re)configure the `repositories` Pod somehow such that it also serves the contents of other directories on the system at some URLs?

**Comment by gdemonet** [comment]
Seems like a reasonable proposal, IMO. One benefit I see is that once we provide high-availability for this service (see #2002), such additional files would also be part of it.

One thing however: I think we should impose some constraints on the allowed custom configurations. For instance, we could ensure the new directories follow the expected package repository structure, and we would manage the NGINX configuration ourselves (no arbitrary includes, unlike what is done for Solutions, for which IMHO we should at the very least validate said includes).

**Reply by NicolasT** [reply]
> For instance, we could ensure the new directories follow the expected package repository structure

Which benefits does that bring? Why couldn't this HTTP server serve... any file tree?

> no arbitrary includes

Fully agree. I'm thinking something along the lines of allowing to configure some dictionary mapping a (sub)route to a path, then have these (sub)routes available under, say, `/static/`, and simply serve files from the given path. No magic, no special cases.
